workshops/build-a-binderhub/workshop-presentations/zero-to-binderhub.md

```diff
@@ -0,0 +1,527 @@
+
+Create a folder named after your BinderHub.
+```bash
+mkdir shf_test_hub
+cd shf_test_hub
+```
+
+We created this folder at the same level as the cluster folder we created in [Step 4: Choose a Cluster Name](#aks-step4) (i.e. in `~/Desktop`).
+
+> **Discussion topic:** Where is a sensible place to keep _this_ folder?
+
+Create two random tokens:
+```bash
+openssl rand -hex 32
+openssl rand -hex 32
+```
+**N.B.:** The command is run **twice** as we need two different tokens.
+
+### 2. Create a `secret.yaml` file <a name="bh-step2"></a>
+ 
+Create a `secret.yaml` file containing the following config and save it in the folder we created in [Step 1: Preparing to Install](#bh-step1).
+```yaml
+jupyterhub:
+  hub:
+    services:
+      binder:
+        apiToken: "<output of FIRST 'openssl rand -hex 32' command>"
+  proxy:
+    secretToken: "<output of SECOND 'openssl rand -hex 32' command>"
+```
+
+To connect to DockerHub, add the following lines.
+```yaml
+registry:
+  username: <docker-id>
+  password: <password>
+```
+**N.B.:** `registry` is on the same level as `jupyterhub`.
+
+> **Discussion topic:** Is it possible to encrypt this information?
+
+Click [here](#secret) for a complete example `secret.yaml` file.
+
+### 3. Create a `config.yaml` file <a name="bh-step3"></a>
+
+Create a `config.yaml` file with the following information and save it in the folder we created in [Step 1: Preparing to Install](#bh-step1).
+
+```yaml
+config:
+  BinderHub:
+    use_registry: true
+    image_prefix: <docker-id>/sheff-binder-
+```
+**N.B.:**
+  * If your Docker account is part of an organisation where you would like to store images instead, change the value of `image_prefix` to `<docker-id|organisation-name>/<prefix>-`
+  * The `<prefix>` can be any string since it will be preppended to image names.
+  It is recommended to be something short and descriptive, such as `binder-dev-` (for development) or `binder-prod-` (for the final product).
+
+### 4. Install BinderHub <a name="bh-step4"></a>
+
+First, pull the latest Helm chart for BinderHub.
+```bash
+helm repo add jupyterhub https://jupyterhub.github.io/helm-chart
+helm repo update
+```
+
+Next, install the required Helm chart using the config files we created in Steps [2: Create a `secret.yaml` file](#bh-step2) and [3: Create a `config.yaml` file](#bh-step3).
+```bash
+helm install jupyterhub/binderhub --version=0.2.0-3b53fce \
+    --name=shfhub \
+    --namespace=shfhub \
+    -f secret.yaml -f config.yaml
+```
+* `--version` refers to the version of the BinderHub Helm Chart.
+  Available versions can be found [here](https://jupyterhub.github.io/helm-chart/#development-releases-binderhub).
+  We have used the version released on March 3rd 2019.
+* `--name` and `--namespace` may be different, but it's recommended they be the same to avoid confusion.
+  It should be something short and descriptive.
```

Comment:
Maybe it should be sheff-hub? It took me some time to realise what shf stood for (but maybe I'm just slow :smile:)

Reply:
Possibly, I couldn't remember which parts azure complained about symbols in namespaces when I wrote it 😄

Reply:
I think I actually got SHF from the train station code 😂